### ISSUE TITLE
Expose debugging macros at build time (and disable by default?)

### ISSUE THREAD

**Issue by orlitzky** [BRiAl/BRiAl#56]
Nothing super important here, but while I was troubleshooting a crash I noticed that some debug macros are enabled by default, e.g.

```C
#if defined(PBORI_NDEBUG)
#define PBORI_ASSERT(arg) (static_cast<void>(0))
#else
#define PBORI_ASSERT(arg) assert(arg)
#endif
```

in `include/polybori/pbori_defs.h`. It might make sense to disable these in release builds, but make them available as a configure flag? My issue was solved by rebuilding brial and turned out not to be related.

**Comment by kiwifb** [comment]
Yes, I am sure we can add some configure option to define `PBORI_NDEBUG`. I'll have a think about it. Something could easily be added to the ebuild as a use-flag right now if we wanted.

**Comment by orlitzky** [comment]
Yeah I was thinking `USE=debug` could toggle an `append-cppflags` right now but it's probably not worth the trouble until there's some other change to brial because we'd have to do a revbump and restabilize it.

**Comment by kiwifb** [comment]
On this side, it is a bit the same, do I cut a release just for that flag? Although considering some stuff I have seen debian side, I probably should do release that dumps the sage-brial folder. That way they would have to get out of packaging it. Still, I should look around if there are any other stuff that needs tweaking to make it worth it.

**Comment by orlitzky** [comment]
> On this side, it is a bit the same, do I cut a release just for that flag?

If I were you, I would definitely wait until there was more.

**Comment by kiwifb** [comment]
Apart from your example above it is mainly
```
#ifndef PBORI_NDEBUG
```
double negative stuff. I think I will switch to a "positive" flag.
```
#ifdef PBORI_DEBUG
```
I am not making a logic error there?

**Comment by orlitzky** [comment]
> I am not making a logic error there?

They're not equivalent obviously but I think it's OK to switch them. Doing so will disable the assertions by default.

**Comment by kiwifb** [comment]
That it will. Which is what you expect of something past the 1.0 release really.

**Comment by kiwifb** [comment]
Master branch now has a `--enable-debug` option.

**Comment by kiwifb** [comment]
Not very happy with the noise generated when compiling with `-Wall` and `--enable-debug` but most of it would need a much better understanding of what the code does to fix. Still, it looks to me that a lot of stuff may not be doing what it is supposed to in this space.

**Comment by dimpase** [comment]
hmm, this starts showing up if I build brial in the homebrew formula, even if I tell it to not do it (?)

```
% ./sage
CErrorInfo()
BoolePolyRing(size_type)
...
```

https://github.com/dimpase/homebrew-tap/pull/31/changes

```ruby     
system "./configure", "--with-boost=#{HOMEBREW_PREFIX}", "--with-boost-libdir=#{HOMEBREW_PREFIX}/lib",
      "--disable-static", "--disable-debug", "--disable-silent-rules", *std_configure_args
 ```
which is to say
```bash
    ./configure --with-boost=#{HOMEBREW_PREFIX} --with-boost-libdir=#{HOMEBREW_PREFIX}/lib
      --disable-static --disable-debug --disable-silent-rules ...
```

Is there a patchless way to tell it to shut up?

**Comment by kiwifb** [comment]
By default it should not be on, which makes me suspect that it is turned on by the bit `*std_configure_args`. So you would need to add your extra argument after it.

**Comment by dimpase** [comment]
`*std_configure_args` does include the `--disable-debug` already. So something is off in this rabbit hole.
Cf. https://docs.brew.sh/Formula-Cookbook

I tried removing `"--disable-silent-rules"`, and this still results in `PBORI_DEBUG` being defined in the header.

**Comment by dimpase** [comment]
No, it's a bug in `configure.ac`. Debugging is always on now, no matter what!

The semantic of `AC_ARG_ENABLE` is not what's done there.

Namely, one should analyse the value of `enableval` and act accordingly, not blindly `[AC_DEFINE([PBORI_DEBUG],[],[enable extra debugging statements])])` or whatever.
@jengelh

**Comment by dimpase** [comment]
Oops, it's actually 8eba4b05b by @kiwifb - not @jengelh (sorry Jan)

**Comment by dimpase** [comment]
OK, I fixed it in 835ed301275d4568d4f148f867f4fcabf3c3346c
